### ISSUE TITLE
install gext System-wide?

### ISSUE THREAD

**Issue by D1Ck3n** [essembeh/gnome-extensions-cli#14]
Good day,

love your tool!

one question i have is: is it possible to install it system-wide? when i install it via pip, it will be installed on /home/username/.local/bin and then i get an warn message that it is not be used with gext alias or for other users.

**Comment by essembeh** [comment]
Hi

### TL;DR: don't do that :)



To be honest I wouldn't recommend to install any package system-wide using `pip`, but it is possible.
To install a package with `pip` on your system, you can run 

```sh
$ sudo pip3 install --system gnome-extensions-cli
```
> Note: I've not tested it but just read the documention (I *never* install packages on my system using `pip`)

Pros:
- you can upgrade the package for all users with a single command like `sudo pip3 install --upgrade --system gnome-extensions-cli`
- you install the package once, you save a couple of MB if you have some users

Cons:
- you have to really trust the authors of the packages you install, because you are root when you install them ...
- installing an/or upgrading system-wide packages as root using `pip` can lead to a mess with versions installed with `apt` (and trust me, you don't want to fix that manually)

The best alternative would be to package `gnome-extensions-cli` as a Debian packages for Debian/Ubuntu so that you can install it and automatically upgrade it with `apt`.
Or even better, Gnome itself should provide a way to install extensions from a terminal :)


As the `gnome-extensions-cli` package itself is not that big, I would really recommend that every user installs its own version. 

Pros:
- you don't mess with system preinstalled packages/version
- you don't give too much privileges to an app not tested/audited by Debian (or whatever distribution you are using) maintainers.

Cons:
- you duplicate some MB on you filesystem.
- every user should periodically run a command to upgrade his version


I hope you'll find the best way for your needs.

**Comment by D1Ck3n** [comment]
thanks for the reply! :) 

i asked because i want to install and enable extensions for other users via ansible. if i install it for my setup user, the extension will not be available for other users, right?

**Comment by essembeh** [comment]
Ah ok, I see.

I don't know ansible, but you should be able to run user commands I guess, so you can manually install it in userland.
You should also be able to install extensions using `gext` as it will use the fallback * filesystem backend* ( [see doc](https://github.com/essembeh/gnome-extensions-cli#filesystem-backend) ), because the default and more reliable backend which uses Dbus needs a running graphical session.

But enabling extensions is another thing. as you can see [here](https://github.com/essembeh/gnome-extensions-cli/blob/main/gnome_extensions_cli/filesystem.py#L119) the filesystem backend relies on `gsettings` command to enable extensions, and I don't know if gsettings works without a running graphical session.

You can let me know if this works, I would update the doc to give some details about your workflow, it might be useful to somebody :+1:

**Comment by D1Ck3n** [comment]
i found my solution :+1: 

installing with sudo makes it system-wide avaiable. I know the risk and will live with it.

this can be closed
